packages/@aws-cdk/aws-cloudtrail/lib/index.ts

```diff
@@ -149,14 +149,13 @@ export class CloudTrail extends cdk.Construct {
       });
       this.cloudWatchLogsGroupArn = logGroup.logGroupArn;
 
-      const logsRole = new iam.Role(this, 'LogsRole', {assumedBy: new iam.ServicePrincipal(cloudTrailPrincipal) });
+      const logsRole = new iam.Role(this, 'LogsRole', { assumedBy: new iam.ServicePrincipal(cloudTrailPrincipal) });
+      this.cloudWatchLogsRoleArn = logsRole.roleArn;
 
       const streamArn = `${this.cloudWatchLogsRoleArn}:log-stream:*`;
```

Comment:
Wait why was this not a static analysis error?

Reply:
The field is `string | undefined` (there is no value if you don't enable CloudWatch/Logs output) - and unassigned it is `undefined`, which is valid. Interpolating `undefined` is totally valid (and it'll output a literal `undefined`).